cli/src/command/generate_config.rs

```diff
@@ -104,30 +102,40 @@ impl GenerateConfig {
     }
 
     /// If no port is provided by user, uses `DEFAULT_EXONUM_LISTEN_PORT`.
-    fn parse_external_address(input: &str) -> Result<SocketAddr, Error> {
-        if let Ok(address) = input.parse() {
-            Ok(address)
-        } else {
-            let ip_address = input.parse()?;
-            Ok(SocketAddr::new(ip_address, DEFAULT_EXONUM_LISTEN_PORT))
+    fn resolve_peer_address(hostname: &str) -> Result<SocketAddr, Error> {
+        match hostname.to_socket_addrs() {
+            Ok(mut addrs) => addrs.next().ok_or_else(|| {
+                Error::msg(format!(
+                    "No one IP address is related to the domain: {}",
+                    hostname
+                ))
+            }),
+            Err(e) => {
+                if e.kind() == ErrorKind::InvalidInput {
+                    Self::resolve_peer_address(&format!(
```

Comment:
This looks like a potential for an infinite recursion. I'd suggest to limit the recursion, e.g., by adding a Boolean function arg (have we appended a default port to the address?).

Reply:
Yeah. The reason I didn't do this is because it can occur only on config generation stage. And to be honest I couldn't think up a case when it happened.